apisix/plugins/openid-connect.lua

```diff
@@ -440,15 +440,30 @@ function _M.rewrite(plugin_conf, ctx)
         conf.timeout = conf.timeout * 1000
     end
 
+    local path = ctx.var.request_uri
+
     if not conf.redirect_uri then
-        conf.redirect_uri = ctx.var.request_uri
+        conf.redirect_uri = path
     end
 
     if not conf.ssl_verify then
         -- openidc use "no" to disable ssl verification
         conf.ssl_verify = "no"
     end
 
+    if path == (conf.logout_path or "/logout") then
+        local discovery, discovery_err =openidc.get_discovery_doc(conf)
+        if discovery_err then
+            core.log.error("OIDC access discovery url failed : ", discovery_err)
+            return 500
```

Comment:
Why 500 here? 500 means `internal error`, we should not return 500 on any occasion

Reply:
The reason for returning 500 is the same as the reason for returning 500 in this step. https://github.com/apache/apisix/blob/master/apisix/plugins/openid-connect.lua#L507-L518。
The `openidc.authenticate` and `get_discovery_doc` functions both get discovered metadata via openidc_ensure_discovered_data.https://github.com/zmartzone/lua-resty-openidc/blob/master/lib/resty/openidc.lua#L580